src/optimize.d

```diff
@@ -796,8 +796,23 @@ extern (C++) Expression Expression_optimize(Expression e, int result, bool keepL
                 return;
             }
             // If e2 *could* have been an integer, make it one.
-            if (e.e2.op == TOKfloat64 && (e.e2.toReal() == cast(sinteger_t)e.e2.toReal()))
-                e.e2 = new IntegerExp(e.loc, e.e2.toInteger(), Type.tint64);
+            if (e.e2.op == TOKfloat64)
+            {
+                version (all)
```

Comment:
Is there some way we could make this `static if (version < firstFixedVersion)` so that it's immediately clear when it can be removed?